public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx

```diff
@@ -0,0 +1,650 @@
+        }
+        this.setState({concepts: results.items, standardConcepts: standardConcepts, top10Results: topResults,
+        loading: false, medlineTerm: medlineTerm, medlinePlusLink: medlinePlusLink});
+  }
+
+
+   getTopConcepts() {
+        const {searchWord, domain: {domain}, measurementTestFilter, measurementOrderFilter} = this.state;
+        const searchRequest = {
+            query: searchWord,
+            domain: domain.toUpperCase(),
+            standardConceptFilter: StandardConceptFilter.STANDARDORCODEIDMATCH,
+            maxResults: 50,
+            minCount: 1,
+            pageNumber: 0,
+            measurementTests: measurementTestFilter ? 1 : 0,
+            measurementOrders: measurementOrderFilter ? 1 : 0
+        };
+        this.fetchConcepts(searchRequest);
+   }
+
+   fetchConcepts(searchRequest: any) {
+           dataBrowserApi().searchConcepts(searchRequest)
+               .then(results => {
+                    this.processSearchResults(results);
+               }).catch(e => {
+                       console.log(e, 'error');
+                       this.setState({loading: false});
+          });
+   }
+
+    handleChange(val) {
+        this.setState({ searchWord: val, loading: true, currentPage: 1, showTopConcepts: true});
+        this.domainTotals(val);
+    }
+
+    backToMain() {
+        if (this.state.searchWord) {
+                localStorage.setItem('searchText', this.state.searchWord);
+        }
+        navigateByUrl('');
+    }
+
+    selectConcept(concept: any) {
+        const {currentPage} = this.state;
+        if (concept && currentPage > 1) {
+                this.setState({selectedConcept: concept, currentPage: 1}, () => {this.getTopConcepts(); });
+        } else {
+                this.setState({selectedConcept: concept});
+        }
+    }
+
+    getTopResultsSize() {
+        const {top10Results, title} = this.state;
+        const top10ResultsTitle = (top10Results.length === 1) ? top10Results.length + ' ' + title.slice(0, -1) :
+        (top10Results.length < 10 ? top10Results.length + ' ' + title : 10 + ' ' + title);
+        return top10ResultsTitle;
+    }
+
+    handlePageClick = (data) => {
+        const {searchWord, domain: {domain}, measurementTestFilter, measurementOrderFilter} = this.state;
+        const searchRequest = {
+            query: searchWord,
+            domain: domain.toUpperCase(),
+            standardConceptFilter: StandardConceptFilter.STANDARDORCODEIDMATCH,
+            maxResults: 50,
+            minCount: 1,
+            pageNumber: data.selected,
+            measurementTests: measurementTestFilter ? 1 : 0,
+            measurementOrders: measurementOrderFilter ? 1 : 0
+        };
+        let showTopConceptsFlag = true;
+        if (data.selected > 0) {
+            showTopConceptsFlag = false;
+        }
+        this.setState({currentPage: data.selected + 1, showTopConcepts: showTopConceptsFlag});
```

Comment:
Nit: can simplify to a single expression and set `showTopConcepts` inline:
```suggestion
        this.setState({currentPage: data.selected + 1, showTopConcepts: data.selected <= 0});
```

Reply:
changed this thanks.